generator/C/include_v1.0/protocol.h

```diff
@@ -186,13 +186,42 @@ static inline void mav_array_memcpy(void *dest, const void *src, size_t n)
 	}
 }
 
+/*
+ * Array direct assignment, for use when fields align and no byte swapping
+ * is required. Most are directly #defined to mav_array_memcpy, except for 
+ * mav_array_assign_char, which uses strncpy instead.
+ */
+#if !MAVLINK_NEED_BYTE_SWAP && MAVLINK_ALIGNED_FIELDS
+static inline void mav_array_assign_char(char *dest, const char *src, size_t n)
+{
+	if (src == NULL) {
+		memset(dest, 0, n);
+	} else {
+		strncpy(dest, src, n);
```

Comment:
the problem is that this will leave the bytes after the string uninitialised, which leaks information and will change the CRC. I think we need to have a mav_strcpy() which zero fills

Reply:
Hi. My understanding of strncpy is that it already will zero-fill the destination buffer up to n, in the case where the src buffer is shorter.
e.g. [this page](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strncpy.html#:~:text=If%20the%20array%20pointed%20to,are%20written.) states "If the array pointed to by s2 is a string that is shorter than n bytes, NUL characters shall be appended to the copy in the array pointed to by s1, until n bytes in all are written."

Reply:
I didn't know that `strncpy` fills with zeros. I also always assumed it just wrote only one zero. In that case, I think this should work.

@tridge do you agree?

Reply:
I'm bumping this as it would be nice to have!

Reply:
hmm, it does seem to be guaranteed by the strncpy definition
```
       If the length of src is less than n, strncpy() writes additional null bytes to dest to ensure that a total of n bytes are written.
```

Reply:
I've added a code comment to each strncpy call to say this, as it is indeed not a very known feature:
`/* strncpy will zero pad dest array up to n bytes */`
Is this then ok to resolve the review comment?
Also rebased to latest master.